include/envoy/network/transport_socket.h

```diff
@@ -12,8 +12,12 @@ namespace Network {
  * Action that should occur on a connection after I/O.
  */
 enum class PostIoAction {
-  // Close the connection.
+  // Remote side sent a half-close.
+  HalfClose,
```

Comment:
`HalfClose` shouldn't be in `PostIoAction` since it is not telling the connection what to do after I/O. The remote sent half-close should be a flag in `IoResult`.

Reply:
Yes, I agree.  In thinking this through today from the other feedback, this makes sense.